bigdecimal/src/lib.rs

```diff
@@ -0,0 +1,637 @@
+        let right = other.clone().set_scale(scale);
+
+        BigDecimal::new(left.int_val - right.int_val, left.scale)
+    }
+}
+
+forward_all_binop_to_ref_ref!(impl Mul for BigDecimal, mul);
+
+impl<'a, 'b> Mul<&'b BigDecimal> for &'a BigDecimal {
+    type Output = BigDecimal;
+
+    #[inline]
+    fn mul(self, other: &BigDecimal) -> BigDecimal {
+        let scale = self.scale + other.scale;
+
+        BigDecimal::new(self.int_val.clone() * other.int_val.clone(), scale)
+    }
+}
+
+forward_all_binop_to_ref_ref!(impl Div for BigDecimal, div);
+
+impl<'a, 'b> Div<&'b BigDecimal> for &'a BigDecimal {
+    type Output = BigDecimal;
+
+    #[inline]
+    #[allow(non_snake_case)]
+    fn div(self, other: &BigDecimal) -> BigDecimal {
+        let scale = self.scale - other.scale;
+        let ref num = self.int_val;
+        let ref den = other.int_val;
+        let (quotient, remainder) = num.div_rem(&den);
+
+        // no remainder - quotient is final solution
+        if remainder == BigInt::zero() {
+            return BigDecimal::new(quotient, scale);
+        }
+
+        let BIG_TEN = &BigInt::from_i8(10).unwrap();
+        let mut remainder = remainder * BIG_TEN;
+        let mut quotient = quotient;
+
+        let MAX_ITERATIONS = 100;
+        let mut iteration_count = 0;
+        while remainder != BigInt::zero() && iteration_count < MAX_ITERATIONS {
+            let (q, r) = remainder.div_rem(&den);
+            quotient = quotient * BIG_TEN + q;
+            remainder = r * BIG_TEN;
+
+            iteration_count += 1;
+        }
+        let scale = scale + iteration_count;
+        BigDecimal::new(quotient, scale)
+    }
+}
+
+forward_all_binop_to_ref_ref!(impl Rem for BigDecimal, rem);
+
+impl<'a, 'b> Rem<&'b BigDecimal> for &'a BigDecimal {
+    type Output = BigDecimal;
+
+    #[inline]
+    fn rem(self, _: &BigDecimal) -> BigDecimal {
+        // let (_, r) = self.div_rem(other);
+        // return r;
+        self.clone()
+    }
+}
+
+impl fmt::Display for BigDecimal {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.pad_integral(true, "", &self.int_val.to_str_radix(10))
+    }
+}
+
+impl Num for BigDecimal {
+    type FromStrRadixErr = ParseBigDecimalError;
+
+    /// Creates and initializes a BigDecimal.
+    #[inline]
+    fn from_str_radix(s: &str, radix: u32) -> Result<BigDecimal, ParseBigDecimalError> {
+        assert!(2 <= radix && radix <= 36, "The radix must be within 2...36");
+
+        let exp_separator: &[_] = &['e', 'E'];
+
+        // split slice into base and exponent parts
+        let (base_part, exponent_value) = match s.find(exp_separator) {
+            // exponent defaults to 0 if (e|E) not found
+            None => (s, 0),
+
+            // split and parse exponent field
+            Some(loc) => {
+                let (base, exp) = s.split_at(loc);
+                // slice after 1 to skip the 'e' char
+                (base, try!(i64::from_str(&exp[1..])))
```

Comment:
We currently still support Rust 1.0, but `split_at` wasn't added/stabilized until 1.4.

But just `s[..loc]` and `s[loc+1..]` should be fine.  (It's mutable slicing where you get into trouble.)